[Core] Stepper Handlers

jsuarezruiz opened the pull request:
### Description of Change ###

Stepper Handlers.

| API | Android | iOS | 
| ----|:--------:|:---:|
| Increment | ✅  | ✅  |  
| Maximum  | ✅  | ✅  |  
| Minimum  | ✅  | ✅  |  
| Value  | ✅  | ✅  |  
| ValueChanged  | ✅  | ✅  |  

### Platforms Affected ### 

- Core
- iOS
- Android

### PR Checklist ###

- [x] Targets the correct branch
- [x]  Include Handler tests
- [ ] Tests are passing (or failures are unrelated)

StephaneDelcroix commented on src/Platform.Handlers/src/Xamarin.Platform.Handlers/Core/IRange.cs:
```diff
@@ -0,0 +1,11 @@
+{
+	public interface IRange : IView
+	{
+		double Minimum { get; }
```
why no setter ?

StephaneDelcroix commented on src/Platform.Handlers/src/Xamarin.Platform.Handlers/Core/IStepper.cs:
```diff
@@ -0,0 +1,7 @@
+{
+	public interface IStepper : IRange
+	{
+		double Increment { get; }
```
why no setter ?

StephaneDelcroix commented on src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/Stepper/StepperHandler.Standard.cs:
```diff
@@ -0,0 +1,14 @@
+{
+	public partial class StepperHandler : AbstractViewHandler<IStepper, object>
+	{
+		protected override object CreateView() => throw new NotImplementedException();
```
shouldn't that throw in the default (base) implementation ?

jsuarezruiz replied:
But it is an abstract method.

StephaneDelcroix commented on src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/Stepper/StepperHandler.cs:
```diff
@@ -0,0 +1,26 @@
+{
+	public partial class StepperHandler
+	{
+		public static PropertyMapper<IStepper> StepperMapper = new PropertyMapper<IStepper>(ViewHandler.ViewMapper)
```
wouldn't it make more sense to define the mapper in StepperHandler.*.cs, that is closer to the implementation. This way we could also _not_ map properties that do not make sense in some platform, and thus have a capability (by code) to figure out if a property is mapped on a platform or not ?

jsuarezruiz replied:
But we repeat the same mapping N times, many times being the same, right?

StephaneDelcroix commented on src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/Stepper/StepperHandler.iOS.cs:
```diff
@@ -0,0 +1,57 @@
+			base.DisposeView(uIStepper);
+		}
+
+		public static void MapMinimum(IViewHandler handler, IStepper slider) => (handler as StepperHandler)?.UpdateMinimum();
```
we should use direct casting, and no null check. if this happen, we'd better fail (also, @PureWeen has another PR to gentrify this)

StephaneDelcroix replied:
also, you can actually set the value directly, without defining an UpdateIncrement method

jsuarezruiz replied:
This has changed a bit since the comment.

jsuarezruiz commented:
I have rebased and moved the macOS implementation to: https://github.com/xamarin/Xamarin.Forms/tree/stepperhandler-mac in order to facilitate the revision and merge of this PR.

jsuarezruiz commented:
Added _DO-NOT-MERGE!!!_ tag until adding handler tests.

PureWeen commented on src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/View/AbstractViewHandler.Android.cs:
```diff
@@ -4,7 +4,7 @@
 namespace Xamarin.Platform.Handlers
 {
-	public partial class AbstractViewHandler<TVirtualView, TNativeView> : IAndroidViewHandler
+	public partial class AbstractViewHandler<TVirtualView, TNativeView> : Java.Lang.Object, IAndroidViewHandler
```
Why do you need to do this? I don't think the AbstractViewHandler should inherit from Java.Lang.Object

jsuarezruiz replied:
To get the handler here (using the View Tag): https://github.com/xamarin/Xamarin.Forms/blob/1008a65edcf3ca8bd4ade01b5c7bf537db3076d1/src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/StepperHandlerManager.cs#L41

Probably we can achieve the same in another way.  Will take a look.

jsuarezruiz replied:
Changed to do in another way.

jsuarezruiz commented:
Moved to https://github.com/dotnet/maui/pull/517
